geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BinAggregatingIterator.scala

```diff
@@ -72,32 +72,54 @@ class PrecomputedBinAggregatingIterator extends BinAggregatingIterator {
     // we only need to decode the parts required for the filter/dedupe/sampling check
     // note: we wouldn't be using precomputed if sample by field wasn't the track id
     decodeBin = if (filter) {
-      setDate = if (isDtgArray) {
+      val dtgIndex = encoding.dtgField.get
+      val geomIndex = encoding.geomField.get
+      val trackIndex = encoding.trackIdField.get
```

Comment:
yeah, once per iterator set-up - they're required as Options in the super class, to pass in to the BinaryOutputEncoder. this class is a special case, b/c it needs to reverse parse them out to filter the row values